### PR TITLE
Add basic tests using pytest. Extend README with testing setup

### PR DESCRIPTION
**Pull request by fzipi**
Signed-off-by: Felipe Zipitria <felipe.zipitria@owasp.org>

Adding pytest so test canbe reproduced easily.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -519,28 +538,33 @@ for f in `ls -1 import/*.conf`; do f=`basename ${f}`; diff import/${f} ~/src/owa
...
+```bash
 ./examples/test_lexer.py /path/to/owasp-modsecurity-crs/rules/REQUEST-920-PROTOCOL-ENFORCEMENT.conf debug
 ```
 
 Run the following command to see how the parser applies language rules to the tokens:
 
-```
-./examples/test_parser.py /path/to/owasp-modsecurity-crs/rules/REQUEST-920-PROTOCOL-ENFORCEMENT.conf debug
+```bash
+./examples/parse.py /path/to/owasp-modsecurity-crs/rules/REQUEST-920-PROTOCOL-ENFORCEMENT.conf debug
```
Comment (airween) [inline README.md]: The mentioned file (`examples/parse.py`) doesn't exists, only the `test_parser`. The parser and lexer are two different steps, should be keep both examples.

File: README.md [hunk 2]
```diff
@@ -1,44 +1,43 @@
...
+# modsec_parser
 
 Welcome to the `modsec_parser` documentation.
 
-Prerequisites
-=============
+The parser runs under Python 3.6+ on Linux, Windows and Mac.
+
+## Installation
+---------------
```
Comment (airween) [inline README.md]: May be we can drop at least `---------------` or the `##`, and use `===============`. But the rendered content is a bit weird with two underlines :).

File: README.md (hunk 1 above)
Comment (fzipi) [reply README.md]: Done.

File: README.md (hunk 2 above)
Comment (fzipi) [reply README.md]: Changed back to `=====`

**Comment by airween** [comment]
Merged.

Thank you so much for this contribution. I'm very pleased to have our test subsystem.

**Pull request merged**